lib/cylc/conditional_simplifier.py

```diff
@@ -19,22 +19,50 @@
 import re
 import ast
 import copy
-import sys
 
 
 class ConditionalSimplifier(object):
     """A class to simplify logical expressions"""
+    RE_CONDITIONALS = re.compile("(&|\||\(|\))")
```

Comment:
The combination backslash escape + pipes (or-logic) + bracket (capture) are making the regular expression very difficult to read. Perhaps better to capture a set in square bracket like this `r'([&|()])'`?